x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/general_logic/basic_license_essentials_tier/ecs_field_duplication.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from 'expect';
+
+import { ALERT_ORIGINAL_TIME } from '@kbn/security-solution-plugin/common/field_maps/field_names';
+import {
+  createRule,
+  deleteAllAlerts,
+  deleteAllRules,
+  getAlertsById,
+  getRuleForAlertTesting,
+  waitForAlertsToBePresent,
+  waitForRuleSuccess,
+} from '../../../../../../../common/utils/security_solution';
+import { getPreviewAlerts, previewRule } from '../../../../utils';
+import { FtrProviderContext } from '../../../../../../ftr_provider_context';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+  const es = getService('es');
+  const log = getService('log');
+
+  describe('@ess @serverless @serverlessQA Source ECS fields copied to other alert fields', () => {
+    before(async () => {
+      await esArchiver.load(
+        'x-pack/test/functional/es_archives/security_solution/ecs_fields_duplicated_for_alerts'
+      );
+    });
+
+    after(async () => {
+      await esArchiver.unload(
+        'x-pack/test/functional/es_archives/security_solution/ecs_fields_duplicated_for_alerts'
+      );
+      await deleteAllAlerts(supertest, log, es);
+      await deleteAllRules(supertest, log);
+    });
+
+    describe('data_stream fields', () => {
+      it('generates alerts with both the original and copied data_stream fields', async () => {
+        const rule = getRuleForAlertTesting(['ecs_fields_duplicated_for_alerts']);
+        const { id } = await createRule(supertest, log, rule);
+        await waitForRuleSuccess({ supertest, log, id });
+        await waitForAlertsToBePresent(supertest, log, 3, [id]);
+        const alertsResponse = await getAlertsById(supertest, log, id);
```

Comment:
might be useful to test filtering by the new `data_stream` fields as well to make sure they get mapped correctly?

Reply:
Good idea, added just such a test in https://github.com/elastic/kibana/pull/220447/commits/cc72a1b38d3c2c6f95ff177a6e6ab0f0ddced31a  👍